src/libAtomVM/externalterm.c

```diff
@@ -577,7 +578,7 @@ static int calculate_heap_usage(const uint8_t *external_term_buf, size_t remaini
                     return INVALID_TERM_SIZE;
                 }
                 u += 1;
-                if (UNLIKELY(remaining < element_size)) {
+                if (UNLIKELY(remaining < (size_t)element_size)) {
```

Comment:
I think all this casts aren't the right thing: I would rather change the code in order to work seamlessly with size_t rather than forcing everytime to size_t using casts.
Also, short coding style comment: `(size_t)foo` -> `(size_t) foo`.

I would suggest to do this change on another commit or maybe PR.